memif-attach/main.go

```diff
@@ -82,24 +133,85 @@ func createMemif(ctx context.Context, conn api.Connection, socketID uint32, mode
 	if err != nil {
 		log.Entry(ctx).Fatalln("ERROR: memifCreate failed:", err)
 	}
-	log.Entry(ctx).Infof("Memif interface created")
-	log.Entry(ctx).Infof("swIfIndex"+
-		"Role"+
-		"SocketID",
+	log.Entry(ctx).Infof("Memif interface created\n"+
+		"swIfIndex %v\n"+
+		"Role %v\n"+
+		"SocketID %v\n",
 		rsp.SwIfIndex, memifCreate.Role, memifCreate.SocketID)
-	// ifindex.Store(ctx, isClient, rsp.SwIfIndex)
-	return nil
+
+	// Set AdminUp
+	swinterface := &interfaces.SwInterfaceSetFlags{
+		SwIfIndex: rsp.SwIfIndex,
+		Flags:     interface_types.IF_STATUS_API_FLAG_ADMIN_UP,
+	}
+	retVal, err := interfaces.NewServiceClient(conn).SwInterfaceSetFlags(ctx, swinterface)
+	if err != nil {
+		log.Entry(ctx).Fatalln("Set AdminUp ERROR:", err)
+	}
+	log.Entry(ctx).Infof("SwInterfaceSetFlags %v", retVal)
 }
 
-func dumpMemif(ctx context.Context, conn api.Connection) (memif.RPCService_MemifDumpClient, error) {
+func dumpMemif(ctx context.Context, conn api.Connection) {
 	c := memif.NewServiceClient(conn)
 
-	MemifDumpClient, MemifDumpErr := c.MemifDump(ctx, &memif.MemifDump{})
-	if MemifDumpErr != nil {
-		log.Entry(ctx).Fatalln("ERROR: MemifDump failed:", MemifDumpErr)
+	memifDumpMsg, err := c.MemifDump(ctx, &memif.MemifDump{})
+	if err != nil {
+		log.Entry(ctx).Fatalln("ERROR: MemifDump failed:", err)
 	}
 	log.Entry(ctx).Infof("Socket file dump")
-	return MemifDumpClient, MemifDumpErr
+	for {
+		reply, err := memifDumpMsg.Recv()
+		if err == io.EOF {
```

Comment:
```suggestion
if err != nil {
```

Reply:
I can not see the logic here. I think when I get the error io.EOF, that should mean the Recv() call has no more stream data to be received.